manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java

```diff
@@ -23,9 +32,45 @@ public void apply(Project project) {
         // add actions to manipulate project
         project.afterEvaluate(new ProjectChangeVersionAction(correspondingModule));
         project.afterEvaluate(new OverrideDependenciesAction(correspondingModule));
-        project.afterEvaluate(new PublishingRepositoryAction());
+        //        project.afterEvaluate(new PublishingRepositoryAction());
         project.afterEvaluate(new ManifestUpdateAction(correspondingModule));
-        project.afterEvaluate(new UploadTaskTransformerAction(correspondingModule));
-        project.afterEvaluate(new PublicationPomTransformerAction(correspondingModule));
+        //        project.afterEvaluate(new UploadTaskTransformerAction(correspondingModule));
+        //        project.afterEvaluate(new PublishingPomTransformerAction(correspondingModule));
+
+        // TODO: add functional tests
+        project.afterEvaluate(new Action<Project>() {
+            @Override
+            public void execute(Project project) {
+                Object preferenceProperty = project.getProperties().get("gm.publishing-plugin");
+                if (preferenceProperty != null) {
+                    log.info("{} property value: {}", "gm.publishing-plugin", preferenceProperty);
+                }
+                if ("maven".equals(preferenceProperty)
+                        && !project.getPluginManager().hasPlugin("maven")) {
+                    log.info("Applyng `maven` plugin.");
+                    project.getPluginManager().apply(MavenPlugin.class);
+                } else if ("maven-publish".equals(preferenceProperty)
+                        && !project.getPluginManager().hasPlugin("maven-publish")) {
+                    log.info("Applyng `maven-publish` plugin.");
+                    project.getPluginManager().apply(MavenPublishPlugin.class);
+                    project.afterEvaluate(new PublishingArtifactsAction());
+                }
+
+                // determine which publishing plugin to use
+                if (project.getPluginManager().hasPlugin("maven")) {
```

Comment:
What happens if the project is already configured with e.g. `maven` - do we need to remove the existing configurations?

Reply:
As I described above, I want to intentionally use the already configured plugin, which simplifies configuration for us. Possibly we may need to remove only the existing publication repository config, but not in Spring case, because it doesn't have any repo configured. In Hibernate I didn't need to do that either.

Reply:
Ok thanks